src/components/Components.js

```diff
@@ -108,45 +109,100 @@ export const RoundButton = ({ icon, size, style, onPress }) => (
   </TouchableOpacity>
 );
 export const RoundTextButton = ({ onPress, style, ...props }) => (
-  <TouchableOpacity onPress={onPress} style={style}>
+  <TouchableOpacity onPress={onPress}>
     <RoundText
       {...props}
-      style={{
-        backgroundColor: PRIMARY_WEAK,
-        elevation: 5,
-        shadowColor: BLACK_TRANSPARENT,
-        shadowOpacity: 1,
-        shadowRadius: 0,
-        shadowOffset: {
-          height: 4,
-          width: 4
+      style={[
+        {
+          backgroundColor: PRIMARY_WEAK,
+          elevation: 5,
+          shadowColor: BLACK_TRANSPARENT,
+          shadowOpacity: 1,
+          shadowRadius: 0,
+          shadowOffset: {
+            height: 4,
+            width: 4
+          },
+          opacity: 1
         },
-        opacity: 1
-      }}
+        style
+      ]}
+      // size={highlighted ? size * 1.2 : size}
     />
   </TouchableOpacity>
 );
 
-export const RoundText = ({ text, size, style, textstyle }) => (
-  <View
-    style={[
-      {
-        backgroundColor: WHITE_TRANSPARENT,
-        borderRadius: 9999,
-        padding: 5,
-        height: size,
-        width: size,
-        alignItems: 'center',
-        justifyContent: 'center'
-      },
-      style
-    ]}
-  >
-    <Text style={[DEFAULT, textstyle]}>
-      {text}
-    </Text>
-  </View>
-);
+export class RoundText extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      size: new Animated.Value(props.size),
+      fontSize: new Animated.Value(this.getBaseFontSize(props))
+    };
+  }
+
+  getBaseFontSize = props =>
+    path(['textStyle', 'fontSize'], props) || DEFAULT.fontSize;
```

Comment:
just to clarify: path is used, because props.textStyle.fontSize fails when textStyle is undefined, yes?